### PR TITLE
Vendor ipaddress and email.utils; renorm ipv4 and ipv6 types based on updated ipaddress corrections (SYN-5490, SYN-7937)

### PR DESCRIPTION
**Pull request by vEpiphyte**

### CODE REVIEW

File: synapse/tests/test_lib_modelrev.py [hunk 1]
```diff
@@ -1170,3 +1170,51 @@ async def test_modelrev_0_2_28(self):
     async def test_modelrev_0_2_29(self):
         async with self.getRegrCore('model-0.2.29') as core:
             self.len(2, await core.nodes('ou:industry:type:taxonomy'))
+
+    async def test_modelrev_0_2_30(self):
+        async with self.getRegrCore('model-0.2.30') as core:
```
Comment (vEpiphyte) [inline synapse/tests/test_lib_modelrev.py:1175]: Relies on https://github.com/vertexproject/synapse-regression/pull/24

**Review by MichaelSquires — APPROVED** [review]

File: changes/03e912acf20e02eb0324b4427f3dd596.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+desc: The inet:rc2822:addr type now rejects malformed inputs which could cause incorrect
```
Comment (MichaelSquires) [inline changes/03e912acf20e02eb0324b4427f3dd596.yaml:2]:
```suggestion
desc: The ``inet:rc2822:addr`` type now rejects malformed inputs which could cause incorrect
```

File: changes/638c91f8ea5f603e32bd82c584910590.yaml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+---
+desc: The inet:ipv4:type and inet:ipv6:type secondary properties now reflect updated
```
Comment (MichaelSquires) [inline changes/638c91f8ea5f603e32bd82c584910590.yaml:2]:
```suggestion
desc: The ``inet:ipv4:type`` and ``inet:ipv6:type`` secondary properties now reflect updated
```

**Comment by MichaelSquires** [comment]
Should we be counting test coverage in vendored code?

File: changes/03e912acf20e02eb0324b4427f3dd596.yaml (hunk 2 above)
Comment (Cisphyx) [reply changes/03e912acf20e02eb0324b4427f3dd596.yaml:2]: Also has a typo of `rc2822` instead of `rfc2822`

**Comment by vEpiphyte** [comment]
@MichaelSquires Let me add their unit tests and make them subclass the VendorTest helper :+1:

File: changes/03e912acf20e02eb0324b4427f3dd596.yaml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+---
+desc: The ``inet:rc2822:addr`` type now rejects malformed inputs which could cause incorrect
```
Comment (Cisphyx) [inline changes/03e912acf20e02eb0324b4427f3dd596.yaml:2]:
```suggestion
desc: The ``inet:rfc2822:addr`` type now rejects malformed inputs which could cause incorrect
```

**Review by Cisphyx — COMMENTED** [review]
Looks like we're picking up the vendor test files in coverage, should we add `synapse/vendor/**/test_*.py` to the omit in `.coveragerc` ?

**Comment by vEpiphyte** [comment]
@Cisphyx I removed some of the tests that we'll never run ( osx 10.7 tests... ) and their associated helpers. I actually don't mind having coverage reporting for those vendored tests as it does give us insight into what behaviors the original authors have / had from their original files.

**Review by Cisphyx — DISMISSED** [review]

**Review by MichaelSquires — APPROVED** [review]

**Comment by vEpiphyte** [comment]
Related cpython issue https://github.com/python/cpython/issues/113171

**Pull request merged**